src/hotspot/cpu/aarch64/universalNativeInvoker_aarch64.cpp

```diff
@@ -217,13 +116,6 @@ RuntimeStub* ProgrammableInvoker::make_native_invoker(BasicType* signature,
 }
 
 void NativeInvokerGenerator::generate() {
-  // we can't use rscratch1 because it is r8, and used by the ABI
-  Register tmp1 = r9;
-  Register tmp2 = r10;
-  assert(!target_uses_register(tmp1->as_VMReg()), "conflict");
-  assert(!target_uses_register(tmp2->as_VMReg()), "conflict");
-  assert(!target_uses_register(rthread->as_VMReg()), "conflict");
```

Comment:
I realized these checks were bogus now that we shuffle arguments as part of the native invoker stub. It is fine for the call to use these registers. Though, if `rthread` is clobbered by the call, we would need to save and restore it (currently that's not the case on any platform though, so we assume it doesn't happen).